### PR TITLE
[5.2] Added fix for last page on pagination query builder being ignored and…

### LINKED ISSUE #8252: Paginator Exceeding Pages
If you have 20 items per page on the paginator and you have 21 items, if someone else comes along and deletes one item leaving only 20, now there's only 1 page if the other person then refreshes they are still on page 2 and the paginator returns no results.

```php
$model->paginate(20);
```

Should the paginator not throw an Exception to catch them being out of bounds, or as most paginators do show the last possible page if there are any results at all.

### PR DESCRIPTION
**Pull request by Sladewill**
… current page being used.

I.e you have 61 items 30 per page, you have 3 pages, one user deletes an item now you have 2 pages of 60 items, another user refreshes on page 3 and they are stuck on a page without any results or pagination links.

Fixes #8252

### CODE REVIEW

File: src/Illuminate/Pagination/AbstractPaginator.php [hunk 1]
```diff
@@ -306,14 +306,17 @@ public static function currentPathResolver(Closure $resolver)
 	 * Resolve the current page or return the default value.
 	 *
 	 * @param  string  $pageName
-	 * @param  int  $default
+	 * @param  int     $default
```
Comment (GrahamCampbell) [inline src/Illuminate/Pagination/AbstractPaginator.php]: Don't align these please.

File: src/Illuminate/Pagination/AbstractPaginator.php [hunk 2]
```diff
@@ -306,14 +306,17 @@ public static function currentPathResolver(Closure $resolver)
...
+	 * @param  int     $lastPage
 	 * @return int
 	 */
-	public static function resolveCurrentPage($pageName = 'page', $default = 1)
+	public static function resolveCurrentPage($pageName = 'page', $default = 1, $lastPage)
```
Comment (GrahamCampbell) [inline src/Illuminate/Pagination/AbstractPaginator.php]: You can't have a parameter without a default following parameters with a default.

File: src/Illuminate/Pagination/AbstractPaginator.php [hunk 3]
```diff
@@ -305,15 +305,18 @@ public static function currentPathResolver(Closure $resolver)
 	/**
 	 * Resolve the current page or return the default value.
 	 *
-	 * @param  string  $pageName
-	 * @param  int  $default
+	 * @param  string $pageName
```
Comment (GrahamCampbell) [inline src/Illuminate/Pagination/AbstractPaginator.php]: Please don't change this

**Comment by JosephSilber** [comment]
But your URL is now wrong, and you have duplicate content. You'll be penalized by Google.

If you really care about this you should really be doing a redirect to the last page.

**Comment by GrahamCampbell** [comment]
@JosephSilber is correct. This isn't a "fix" as such.

**Comment by Sladewill** [comment]
What about if it throws an exception instead then at least the user can decide on where they want the user to go or if they want to show an error page?

**Pull request closed without merging**